packages/compartment-mapper/src/node-modules.js

```diff
@@ -269,6 +269,20 @@ const graphPackage = async (
   /** @type {Record<string, Language>} */
   const types = {};
 
+  const readDescriptorUpwards = async path => {
+    const pathSegments = path.split('/');
+    let desc;
+    do {
+      // eslint-disable-next-line no-await-in-loop
+      desc = await readDescriptor(
+        resolveLocation(pathSegments.join('/'), packageLocation),
+      );
+      pathSegments.pop();
+    } while (!desc && pathSegments.length !== 0);
+
+    return desc;
+  };
```

Comment:
Please look at `search.js` and either borrow the technique or factor the code so you can use the same algorithm. This should be implemented in terms of proper URL math for platform independence.

There should be at least one test that would fail on Windows with this algorithm, by using back-slashes.

Reply:
done. refactored search.js a bit to expose the logic of searching but allow using the local readDescriptor function